Documentation/RepoCompose.md

```diff
@@ -0,0 +1,289 @@
+
+``` json
+{
+    "os-windows": {
+        "nuget": { },
+        "file": { }
+    },
+    "os-linux": {
+        "nuget": { },
+        "file": { }
+    }
+}
+```
+
+A ful sample output for `produces` is available in the Samples section.
+
+### change
+
+The `change` command is used to alter the floating build dependencies section.  It can establish new versions of NuGet packages, new locations to find file artifacts, different NuGet feeds, etc ...  
+
+This is the command which allow us to use the build output of one repo as the input of a dependent repo.  The first repo can build using a new output version (say beta5) and dependent repos can be changed to accept this new version.  
+
+This command operates by providing json whose format is a subset of the output of `consumes`.  In particular it will provide the `"dependencies.floating"` section.  
+
+``` json
+"dependencies" {
+    "floating": { 
+        "nuget": {
+            "packages" {
+                "MicroBuild.Core": "0.2.0",
+                "Microsoft.NETCore.Platforms": "1.0.1"
+            }
+        }
+    }
+}
+```
+
+A tool responsible for composing repos would ideally:
+
+1. Execute `run.cmd consumes` and capture the output.
+2. Alter the NuGet package versions in the json to have the correct build identifier: beta5, RTM, etc ..
+3. Execute `run.cmd change` and pass in the altered output.
+
+
+### publish
+
+The `publish` command takes a json input that describes the locations artifacts should be published to.  The input to this command is the output of `produces` that is augmented with location information.  
+
+## Artifact Specification
+
+The json describing artifacts is the same between the `consume`, `produces` and `publish` commands.  These items can be used anywhere artifacts are listed above.
+
+### NuGet packages
+
+The description for NuGet artifacts is it two parts:
+
+1. The set of feeds packages are being read from.
+2. The set of packages that are being consumed and their respective versions.
+
+Example:
+
+``` json
+"nuget": {
+    "feeds": [
+        { 
+           "name": "core-clr",
+           "value": "https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json" 
+        },
+        {
+            "name": "dotnet-core",
+            "value": "https://dotnet.myget.org/F/dotnet-core/api/v3/index.json"
+        }
+    ],
+    "packages": {
+        "MicroBuild.Core": "0.2.0",
+        "Microsoft.NETCore.Platforms": "1.0.1"
+    }
+}
+```
+
+### File 
+
+Any file which is not a NuGet package should be listed as a file artifact.  These can be downloaded from the web or copied from local places on the hard drive.  Each type of file entry will have a name uniquely identifying the artifact and a kind property specifying the remainder of the properties:
+
+    - uri: a property named `"uri"` will contain an absolute Uri for the artifact.
+    - filesystem: a property named `"location"` will contain an OS specific file path for the artifact.
```

Comment:
Is the kind property ("uri" or "filesystem") and its associated property ("uri" or "location"), redundant?  Could you get away with just having a "uri" property or a "filesystem" property and getting rid of the "kind" property?